lib/wallaby/feature_case.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Wallaby.FeatureCase do
+  use ExUnit.CaseTemplate
+
+  @otp_app Application.get_env(:wallaby, :otp_app)
+
+  using do
+    quote do
+      ExUnit.Case.register_attribute(__MODULE__, :sessions)
+
+      use Wallaby.DSL
+      import Wallaby.FeatureCase
+    end
+  end
+
+  setup context do
+    sessions_count = get_in(context, [:registered, :sessions]) || 1
+    capabilities = Application.get_env(:wallaby, :capabilities)
+
+    metadata =
+      @otp_app
+      |> ecto_repos()
+      |> Enum.map(&checkout_ecto_repos(&1, context[:async]))
+      |> metadata_for_ecto_repos()
+
+    sessions =
+      1..sessions_count
+      |> Enum.map(fn _ ->
+        context = String.replace(to_string(context.test), " ", "_")
+
+        {:ok, session} =
+          Wallaby.start_session(context: context, capabilities: capabilities, metadata: metadata)
+
+        session
+      end)
+
+    [sessions: sessions]
+  end
+
+  defp ecto_repos(nil), do: []
+  defp ecto_repos(otp_app), do: Application.get_env(otp_app, :ecto_repos, [])
+
+  defp checkout_ecto_repos(repo, async) do
+    :ok = Ecto.Adapters.SQL.Sandbox.checkout(repo)
+
+    unless async, do: Ecto.Adapters.SQL.Sandbox.mode(repo, {:shared, self()})
+
+    repo
+  end
+
+  def metadata_for_ecto_repos([]), do: Map.new()
+  def metadata_for_ecto_repos(repos), do: Phoenix.Ecto.SQL.Sandbox.metadata_for(repos, self())
+
+  defmacro feature(message, context \\ quote(do: _), contents) do
```

Comment:
There is the potential to make this compatible with the following syntax:

```elixir
feature "some feature test" do
  raise "oopsie"
rescue
  # stuff
end
```

the `property` macro from stream_data has the functionality, I'm not sure how useful it would be to have in wallaby.